services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentMigratorITest.kt

```diff
@@ -0,0 +1,146 @@
+import fi.hel.haitaton.hanke.attachment.common.TestFile
+import fi.hel.haitaton.hanke.factory.ApplicationAttachmentFactory
+import io.mockk.checkUnnecessaryStub
+import io.mockk.clearAllMocks
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.security.test.context.support.WithMockUser
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.transaction.annotation.Transactional
+
+@SpringBootTest
+@ActiveProfiles("test")
+@WithMockUser(USERNAME)
+class ApplicationAttachmentMigratorITest(
+    @Autowired private val attachmentMigrator: ApplicationAttachmentMigrator,
+    @Autowired private val attachmentRepository: ApplicationAttachmentRepository,
+    @Autowired private val contentRepository: ApplicationAttachmentContentRepository,
+    @Autowired private val attachmentFactory: ApplicationAttachmentFactory,
+    @Autowired private val fileClient: MockFileClient,
+) : DatabaseTest() {
+
+    @BeforeEach
+    fun setup() {
+        clearAllMocks()
+        fileClient.recreateContainers()
+    }
+
+    @AfterEach
+    fun tearDown() {
+        checkUnnecessaryStub()
+    }
+
+    @Nested
+    @Transactional
+    inner class FindAttachmentWithDatabaseContent {
+        @Test
+        fun `Should return null if no attachments in content table`() {
+            val attachmentWithContent = attachmentMigrator.findAttachmentWithDatabaseContent()
+
+            assertThat(attachmentWithContent).isNull()
+        }
+
+        @Test
+        fun `Should return un-migrated content when there are any`() {
+            val attachment = attachmentFactory.save().withDbContent().value
+            assertThat(attachmentRepository.findAll()).hasSize(1)
+            assertThat(contentRepository.findAll()).hasSize(1)
+            val content = contentRepository.findByIdOrNull(attachment.id!!)!!
+            assertThat(fileClient.listBlobs(Container.HAKEMUS_LIITTEET)).isEmpty()
+
+            val attachmentWithContent = attachmentMigrator.findAttachmentWithDatabaseContent()
+
+            assertThat(attachmentWithContent).isNotNull().all {
+                prop(ApplicationAttachmentWithContent::id).isEqualTo(attachment.id!!)
+                prop(ApplicationAttachmentWithContent::applicationId)
+                    .isEqualTo(attachment.applicationId)
+                prop(ApplicationAttachmentWithContent::content).all {
+                    prop(AttachmentContent::fileName).isEqualTo(attachment.fileName)
+                    prop(AttachmentContent::contentType).isEqualTo(attachment.contentType)
+                    prop(AttachmentContent::bytes).isEqualTo(content.content)
```

Comment:
Couldn't this check against the data the factory uses?
```suggestion
                    prop(AttachmentContent::bytes).isEqualTo(DEFAULT_DATA)
```

If we want it to be explicit, we can specify the data the factory should use.
```kotlin
val attachment = attachmentFactory.save().withDbContent(DEFAULT_DATA).value
```

Either way, we should know what the content is supposed to be without getting it from the DB.

Reply:
Fixed.